lib/Client.js

```diff
@@ -418,7 +424,7 @@ class Client extends EventEmitter {
     */
     async connect() {
         try {
-            const data = await (this.options.maxShards === "auto" ? this.getBotGateway() : this.getGateway());
+            const data = await (this.options.maxShards === "auto" || (this.options.shardConcurrency === "auto" && this.bot) ? this.getBotGateway() : this.getGateway());
```

Comment:
this won't actually try concurrent sharding if the token isn't manually prefixed

Reply:
Sounds like a reasonable compromise given that:
- `concurrency: 1` is default behavior
- concurrency > 1 is a fairly exclusive permission
- `getBotGateway()` is restricted to bots only
- Forcing manual token prefixing is under discussion